WordPress/gutenberg
# Compose: Introduce in-house `compose` and `pipe` utils

## What?
Currently, we expose our own `compose()` high-order composition utility, which is a re-export of `_.flowRight()`. This PR implements our simple in-house version of `compose()`, and uses the opportunity to also introduce an in-house version of `flow()`, which is the inverted version of `flowRight()`.  That allows us to get rid of the corresponding Lodash functions in the following PRs.

## Why?
Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?
We're implementing a custom base `baseFlow()` function and using that to generate both `compose()` and `flow()` so they work the same way as the Lodash versions. We're adding a few tests to ensure things work as expected.

I've chosen to migrate away from the existing `_.flowRight()` and `_.flow()` in a separate PR, to keep this one more contained.

## Testing Instructions
* Smoke test all editors - `compose()` is used all over the place.
* Verify all tests pass.
